elastic/kibana
# [EPM] Don't crash on invalid characters in integration search bar

## Summary

Fixes https://github.com/elastic/kibana/issues/63834

The error was caused by passing the `queryText` string back to `EuiSearchBar`'s `query` parameter instead of the full `Query` object. This lead to parse errors due to invalid characters crash the page, instead of being nicely handled by `EuiSearchBar` itself and passed to `onChange`. I didn't investigate further why `EuiSearchBar` crashes when that happens.

During implementation I decided not to show the parse error at all, (cf. https://github.com/elastic/kibana/issues/63834#issuecomment-624605222 ) but just hide it. That way, when the user enters invalid characters like `,` or `"`, the list of packages shown just doesn't change.

![image](https://user-images.githubusercontent.com/189073/81420573-72f08d00-9150-11ea-832a-cd4614730294.png)

I can add a nice error box if it is desired, but that would make the package grid jump around, or I need a better place for the error than below the search box.

## How to test this

* Navigate to the `Integrations` tab.
* Enter `,` or `"` in the search bar. Verify that the page continues to work.
* Perform a search with a valid string (like `system`). Verify that the search works and only matching integrations are shown.
